plasma_framework/contracts/src/exits/payment/controllers/PaymentProcessInFlightExit.sol

```diff
@@ -259,6 +247,50 @@ library PaymentProcessInFlightExit {
         framework.batchFlagOutputsFinalized(outputIdsToFlag);
     }
 
+    function returnInputPiggybackBonds(
+        Controller memory self,
+        PaymentExitDataModel.InFlightExit storage exit,
+        address token
+    )
+        private
+    {
+        for (uint16 i = 0; i < PaymentTransactionModel.MAX_INPUT_NUM(); i++) {
+            PaymentExitDataModel.WithdrawData memory withdrawal = exit.inputs[i];
+            if (token == withdrawal.token && exit.isInputPiggybacked(i)) {
```

Comment:
this might be up to the wording (and possibly require a clarifying comment), but I'm assuming here that invalid and challenged piggybacks **won't** return the bond, right?

Reply:
Thanks for pointing this out. I just double checked the challenge piggyback code. So if it challenged, it would return the bond immediately and delete the piggyback flag. So this implementation is fine.

Reply:
But I definitely have the wrong impression in my mind when making the code change....I thought challenge piggyback would end up changing the piggyback bond owner.

Reply:
OK, maybe consider leaving a comment about challenged piggybacks - that this is taken care of b/c `isInputPiggybacked` returns `false`.